### PR TITLE
test: Fix and clean up `smoke-billing-activity.spec.ts` and `DatabaseBackups.test.tsx`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

-  Cleaned up `DatabaseBackups.test.tsx` to hopefully finally stop flake 🧹 
   - I cleaned up the tests as much as possible (for example removed unnecessary mocks)
   - Fixed some flawed tests
   - Separated DBaaS v1 and V2
   - Removed assertions known to be flakey
- Fixed up `smoke-billing-activity.spec.ts` after @abailly-akamai patched it 🧹 
  - Added some extra assertions
  - Re-added dynamic timezone label
  - Fixes incorrect `key` for `Coordinated Universal Time`

## How to test 🧪

```sh
yarn cy:run -s cypress/e2e/core/billing/smoke-billing-activity.spec.ts 
```

```sh
yarn workspace linode-manager test DatabaseBackups
```

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

### CODE REVIEW

File: packages/manager/cypress/e2e/core/billing/smoke-billing-activity.spec.ts [hunk 1]
```diff
@@ -329,11 +329,14 @@ describe('Billing Activity Feed', () => {
     // Time zones against which to verify invoice and payment dates.
     const timeZonesList = [
       { key: 'America/New_York', human: 'Eastern Time - New York' },
-      { key: 'GMT', human: 'Coordinated Universal Time' },
+      { key: 'UTC', human: 'Coordinated Universal Time' },
```
Comment (bnussman-akamai) [inline packages/manager/cypress/e2e/core/billing/smoke-billing-activity.spec.ts:332]: This isn't related to the failures we saw but I noticed this was wrong

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx [hunk 2]
```diff
@@ -32,20 +25,12 @@ describe('Database Backups', () => {
       })
     );
 
-    const { findByText, getAllByRole } = renderWithTheme(<DatabaseBackups />);
-
-    const backupPromises = backups.map((backup) => {
-      const expectedDate = formatDate(backup.created, { timezone: 'utc' });
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx:38]: I just went ahead and removed the assertions that check for the dates. Now, this test will verify 7 backups rows render. Just want to put this flake behind us 😮‍💨

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx [hunk 3]
```diff
@@ -107,9 +91,6 @@ describe('Database Backups', () => {
     const backups = databaseBackupFactory.buildList(7);
 
     server.use(
-      http.get('*/profile', () => {
-        return HttpResponse.json(profileFactory.build({ timezone: 'utc' }));
-      }),
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx:112]: No need to mock the profile. The test doesn't make any assertions about times or dates.

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx [hunk 4]
```diff
@@ -170,33 +146,30 @@ describe('Database Backups', () => {
       })
     );
 
-    const rendered = renderWithTheme(<DatabaseBackups disabled={false} />);
-    expect(
-      rendered.container.getElementsByClassName('MuiDateCalendar-root')
-    ).toBeDefined();
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx:176]: This was flawed logic. `rendered.container.getElementsByClassName` always returns a defined array

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx [hunk 5]
```diff
@@ -1,18 +1,14 @@
...
+import { databaseBackupFactory, databaseFactory } from 'src/factories';
 import { makeResourcePage } from 'src/mocks/serverHandlers';
 import { HttpResponse, http, server } from 'src/mocks/testServer';
-import { formatDate } from 'src/utilities/formatDate';
 import { renderWithTheme } from 'src/utilities/testHelpers';
 
 import DatabaseBackups from './DatabaseBackups';
 
-describe('Database Backups', () => {
+describe('Database Backups (Legacy)', () => {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseBackups/DatabaseBackups.test.tsx:11]: I separated the DBaaS v1 and v2 tests into separate `describe` blocks to make things more clear and easy to clean up

**Review by jdamore-linode — APPROVED** [review]
Thanks @bnussman-akamai!

**Pull request merged**